finish the fifth part DeleteTodoItem.

JungDaewook opened the pull request:
made a new function named removeTodoItem
modified js file's name to functionsOfTodoList because it has not only function addTodoItem but also other functions of this project

Commit:
finish the fifth part DeleteTodoItem.
made a new function named removeTodoItem
modified js file's name to functionsOfTodoList because it has not only function addTodoItem but also other functions of this project

unpo88 commented on functionsOfTodoList.js:
```diff
@@ -0,0 +1,18 @@
+}
+
+function removeTodoItem(todoItemCount) {
+  const DeleteTodoItem = document.getElementById('todoItem' + todoItemCount);
```
@JungDaewook 

변수명을 `addTodoItem`에서는 `lowerCamelCase`를 사용했는데, 여기서는 `UpperCamelCase`를 사용한 이유가 있을까요?

JungDaewook replied:
아무생각 없이 변수명을 선정한 것 같음. 'DeleteTodoItem'이라는 변수는 lowerCamelCase인 'deleteTodoItem'으로 변경하였고 삭제를 위해 선언했던 'todoItemCount'라는 전역변수의 변수명을 UpperCamelCase인 'TodoItemCount'라는 변수명으로 변경함. 지역 변수와 전역 변수를 구분하기 위해서.

unpo88 commented on functionsOfTodoList.js:
```diff
@@ -0,0 +1,18 @@
+  todoItem.innerHTML += "<button type = 'button' onclick = 'removeTodoItem("+todoItemCount+")')> 삭제 </button>";
+  todoList = document.getElementById('todoList');
+  todoList.appendChild(todoItem);
+  todoItemCount++;
```
@JungDaewook 

**[질문]**
이 부분도 가독성을 좋게 하려면 코드를 어떻게 바꾸는 것이 좋을까요? 🤔

JungDaewook replied:
가독성을 위하여 함수 내부의 변수 선언, 'todoItem'안에 입력받은 텍스트를 넣는 과정, 'todoList'안에 'todoItem'을 넣는 과정을 줄바꿈으로 구분함.

unpo88 reviewed (changes requested):
@JungDaewook 

코멘트 남겨놓았습니다.

------------

[질문]

[이전 PR](https://github.com/JungDaewook/Todo-JavaScript/pull/7#pullrequestreview-929030421)에서 다음과 같이 코멘트를 남겨놓았는데요.

> input 에 값을 입력하고, 추가를 눌렀을 때, input 값이 초기화되지 않고, 그대로 유지되는 경우가 있어요. 요것을 초기화해주려면 어떻게 해야할까요? 🤔

해당 작업은 별도의 PR에서 진행할 예정일까요?

----------------

[질문]

[Comment](https://github.com/JungDaewook/Todo-JavaScript/pull/7#discussion_r840652518)에 남겨놓은 것 처럼 [요구사항 문서](https://www.notion.so/Todo-List-for-JavaScript-45dc9e41fe4240b48cd3f0ee624dc5c3)를 보면 다음과 같은 요구사항이 명시되어 있어요.

> - “할 일 목록"을 보여줄 수 있는 리스트가 존재한다.
>     - 할 일 목록은 최대 20개까지 입력 가능하다.

해당 기능은 추후에 별도로 구현될 예정일까요?

----------------

[질문]

1. `PR`의 `Description`을 영어로 설명해놓은 이유가 있나요?
2. `PR`의 `Description`에는 어떤 내용을 담는 것이 좋을까요? 🤔

Commit:
modified the first and second comment
1. DeleteTodoItem이라는 변수명을 deleteTodoItem으로 변경 (lowerCamelCase)
1-1. todoItemCount라는 변수명을 TodoItemCount(UpperCamelCase)로 변경 전역 변수라 지역 변수와 구분이 필요할 것이라고 생각하여 변경함.
2. 가독성을 위하여 addTodoItem이라는 함수에서 변수선언부분과 todoItem에 입력받은 텍스트를 추가해주는 부분, todoList에 todoItem을 추가해주는 부분을 가독성을 위해 줄바꿈으로 구분.

JungDaewook commented:
> [질문]
> 
> [이전 PR](https://github.com/JungDaewook/Todo-JavaScript/pull/7#pullrequestreview-929030421)에서 다음과 같이 코멘트를 남겨놓았는데요.
> 
> > input 에 값을 입력하고, 추가를 눌렀을 때, input 값이 초기화되지 않고, 그대로 유지되는 경우가 있어요. 요것을 초기화해주려면 어떻게 해야할까요? 🤔
> 
> 해당 작업은 별도의 PR에서 진행할 예정일까요?

[답변]
초기화되지 않는 상황에 대하여 인지하였고, 해당 작업은 modifyThatImissed에서 진행할 예정입니다.

JungDaewook commented:
> [질문]
> 
> [Comment](https://github.com/JungDaewook/Todo-JavaScript/pull/7#discussion_r840652518)에 남겨놓은 것 처럼 [요구사항 문서](https://www.notion.so/Todo-List-for-JavaScript-45dc9e41fe4240b48cd3f0ee624dc5c3)를 보면 다음과 같은 요구사항이 명시되어 있어요.
> 
> > * “할 일 목록"을 보여줄 수 있는 리스트가 존재한다.
> >   
> >   * 할 일 목록은 최대 20개까지 입력 가능하다.
> 
> 해당 기능은 추후에 별도로 구현될 예정일까요?

[답변]
목록 안에 20개의 항목만 넣을 수 있도록 modifyThatImissed 브랜치에서 구현하였으나 코멘트를 남겨주셔서 다시 수정해서 해당 브랜치에서 해결할 예정이에요.

JungDaewook commented:
> [질문]
> 
> 1. `PR`의 `Description`을 영어로 설명해놓은 이유가 있나요?
> 2. `PR`의 `Description`에는 어떤 내용을 담는 것이 좋을까요? 🤔

[답변]
1. 커밋 메시지를 공부할 때 영어면 좋다고 생각해서 pr도 영어로 해야지 단순하게 생각했어요. 영어로 적는 경우는 개인적인 감정이 들어갈 것 같은 경우에 그렇게 한다고 해서 다음부터는 한글로 하려고 해요.
2. 'description'에는 무엇이 바뀌었는지, 무엇을 추가했는지, 그리고 무슨 어려움을 겪었고 무슨 고민을 했는지를 담으라는 피드백을 받아서 그렇게 해 볼 예정이에요.

JungDaewook commented:
[기타]
pull request가 닫혔는데 질문에 대한 코멘트를 남기다가 close with comment 버튼을 잘못 눌러서 닫았다가 다시 열었어요.

unpo88 commented on functionsOfTodoList.js:
```diff
@@ -0,0 +1,20 @@
+let TodoItemCount = 1;
+
+function addTodoItem()  {
+  const todoInputText = document.getElementById('todoInputText').value;
```
@JungDaewook 

[선택/제안]

지금 코드를 보다보니까 `todoInputTextValue` 라는 네이밍이 조금 더 직관적일 수도 있겠네요!

@JungDaewook 이 판단해서 작업해주세요!

unpo88 reviewed (approved):
LGTM!

코멘트 하나 남겨진것만 확인 부탁드립니다 🙏

JungDaewook commented:
[선택/제안]

지금 코드를 보다보니까 todoInputTextValue 라는 네이밍이 조금 더 직관적일 수도 있겠네요!

@JungDaewook 이 판단해서 작업해주세요!

[답변/추후 반영 예정]

modifyThatImissed 브랜치에서 다른 부분들과 같이 수정하도록 할게요! 이제 merge해도 될까요?

unpo88 reviewed (approved):
Good~
